### ISSUE TITLE
`istioctl verify-install` fails on parsing error

### ISSUE THREAD

**Issue by surajssd** [istio/istio#25194]
**Bug description**

`istioctl verify-install` reports parsing error `Error: could not load IstioOperator from cluster: could not unmarshal: unknown value "HEALTHY" for enum istio.operator.v1alpha1.InstallStatus_Status`.

**Affected product area**

[x] Installation
[x] User Experience

**Steps to reproduce the bug**

Followed the steps from https://istio.io/latest/docs/setup/install/standalone-operator/

```console
$ istioctl operator init
Using operator Deployment image: docker.io/istio/operator:1.6.4
✔ Istio operator installed                                                                                                                                    
✔ Installation complete
```

```console
$ kubectl create ns istio-system
namespace/istio-system created
```

```console
$ kubectl apply -f - <<EOF
> apiVersion: install.istio.io/v1alpha1
> kind: IstioOperator
> metadata:
>   namespace: istio-system
>   name: example-istiocontrolplane
> spec:
>   profile: minimal
> EOF
istiooperator.install.istio.io/example-istiocontrolplane created
```

And here it fails as:

```json
$ istioctl verify-install 
Error: could not load IstioOperator from cluster: could not unmarshal: unknown value "HEALTHY" for enum istio.operator.v1alpha1.InstallStatus_Status

YAML:
{"apiVersion":"install.istio.io/v1alpha1","kind":"IstioOperator","metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"install.istio.io/v1alpha1\",\"kind\":\"IstioOperator\",\"metadata\":{\"annotations\":{},\"name\":\"example-istiocontrolplane\",\"namespace\":\"istio-system\"},\"spec\":{\"profile\":\"minimal\"}}\n"},"finalizers":["istio-finalizer.install.istio.io"],"generation":1,"managedFields":[{"apiVersion":"install.istio.io/v1alpha1","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}}},"f:spec":{".":{},"f:profile":{}}},"manager":"kubectl","operation":"Update","time":"2020-07-03T12:42:31Z"},{"apiVersion":"install.istio.io/v1alpha1","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:finalizers":{".":{},"v:\"istio-finalizer.install.istio.io\"":{}}},"f:status":{".":{},"f:componentStatus":{".":{},"f:Base":{".":{},"f:status":{}},"f:Pilot":{".":{},"f:status":{}}},"f:status":{}}},"manager":"operator","operation":"Update","time":"2020-07-03T12:43:30Z"}],"name":"example-istiocontrolplane","namespace":"istio-system","resourceVersion":"1068","selfLink":"/apis/install.istio.io/v1alpha1/namespaces/istio-system/istiooperators/example-istiocontrolplane","uid":"128b61af-f7f3-41ab-96d1-aa9538401215"},"spec":{"profile":"minimal"},"status":{"componentStatus":{"Base":{"status":"HEALTHY"},"Pilot":{"status":"HEALTHY"}},"status":"HEALTHY"}}.  Use --filename
```

**Expected behavior**

`istioctl verify-install` works fine and does not report parsing error.

**Version**

```console
$ istioctl version --remote
client version: 1.6.4
control plane version: 1.6.4
data plane version: none
```

```json
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:52:00Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.4", GitCommit:"c96aede7b5205121079932896c4ad89bb93260af", GitTreeState:"clean", BuildDate:"2020-06-17T11:33:59Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**

It is installed by downloading https://github.com/istio/istio/releases/download/1.6.4/istio-1.6.4-linux-amd64.tar.gz.

**Environment where bug was observed**

minikube

PFA the cluster state archive: [istio-dump.tar.gz](https://github.com/istio/istio/files/4869770/istio-dump.tar.gz)

**Comment by esnible** [comment]
Verified this happens on Istio 1.6.x, but not 1.5.x nor on master.

**Comment by morvencao** [comment]
This should be likely relevant to the api change, I can't reproduce this in master branch.

**Comment by shamsher31** [comment]
I was able to reproduce it on `master` with `Istio 1.6.4` but with a slightly different errors. Possibly similar to https://github.com/istio/istio/issues/22280 

```
$ $GOPATH/src/istio.io/istio/out/linux_amd64/istioctl verify-install
Error: could not load IstioOperator from cluster: could not unmarshal: json: cannot unmarshal string into Go value of type map[string]json.RawMessage

YAML:
{"apiVersion":"install.istio.io/v1alpha1","kind":"IstioOperator","metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"install.istio.io/v1alpha1\",\"kind\":\"IstioOperator\",\"metadata\":{\"annotations\":{},\"name\":\"example-istiocontrolplane\",\"namespace\":\"istio-system\"},\"spec\":{\"profile\":\"demo\"}}\n"},"finalizers":["istio-finalizer.install.istio.io"],"generation":1,"managedFields":[{"apiVersion":"install.istio.io/v1alpha1","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}}},"f:spec":{".":{},"f:profile":{}}},"manager":"kubectl","operation":"Update","time":"2020-07-07T09:55:32Z"},{"apiVersion":"install.istio.io/v1alpha1","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:finalizers":{".":{},"v:\"istio-finalizer.install.istio.io\"":{}}},"f:status":{".":{},"f:componentStatus":{".":{},"f:AddonComponents":{".":{},"f:status":{}},"f:Base":{".":{},"f:status":{}},"f:EgressGateways":{".":{},"f:status":{}},"f:IngressGateways":{".":{},"f:status":{}},"f:Pilot":{".":{},"f:status":{}}},"f:status":{}}},"manager":"operator","operation":"Update","time":"2020-07-07T09:56:30Z"}],"name":"example-istiocontrolplane","namespace":"istio-system","resourceVersion":"55487","selfLink":"/apis/install.istio.io/v1alpha1/namespaces/istio-system/istiooperators/example-istiocontrolplane","uid":"1e398c75-4a73-410f-b4f8-23d3a4ddba1d"},"spec":{"profile":"demo"},"status":{"componentStatus":{"AddonComponents":{"status":"HEALTHY"},"Base":{"status":"HEALTHY"},"EgressGateways":{"status":"HEALTHY"},"IngressGateways":{"status":"HEALTHY"},"Pilot":{"status":"HEALTHY"}},"status":"HEALTHY"}}.  Use --filename
```

**Comment by esnible** [comment]
I assigned this to myself but I will give it to anyone who has a plan to fix it.

Verify-install is supposed to read an IOP, generate a deployment to an in-memory struct, and pass that deployment to a function that checks if the deployment is running.  It would benefit from using the logic in the operator instead of using its own logic.

Probably there is no integration test for `verify-install`.  Do the `manifest` commands have integration tests?

**Comment by esnible** [comment]
@shamsher31 this is now happening on `Master`

The error message is particularly poor.  It starts with `Error: could not load IstioOperator from cluster: could not unmarshal: unknown field "citadel" in v1alpha1.IstioComponentSetSpec` and then prints 500 lines of YAML.

**Comment by esnible** [comment]
@ostromart PR https://github.com/istio/api/pull/1464 broke this.

Looking into a work-around.  Was components.citadel ever deprecated or did it go immediately to deleted?

**Comment by tillig** [comment]
I had a _very_ similar problem recently. Doing this...

`istioctl verify-install --revision v1-17-2 --log_output_level all:debug --log_stacktrace_level all:debug`

...I was able to see the stack trace of the thing that was failing, and, long story short, it was when getting the set of Istio operators and trying to deserialize the YAML from those operators.

Doing this...

`kubectl get istiooperators -A`

I found that, for some reason, **I had a couple of super old Istio operators hanging out**, like an uninstall operation had not run cleanly.

After removing the old Istio operators from the system, the `istioctl verify-install` command started working again.
